src/jdk.jdi/share/classes/com/sun/jdi/ThreadReference.java

```diff
@@ -402,8 +402,8 @@ List<MonitorInfo> ownedMonitorsAndFrames()
      * @throws java.lang.IllegalArgumentException if <CODE>frame</CODE>
      * is not on this thread's call stack.
      *
-     * @throws OpaqueFrameException if this thread is a suspended virtual thread and the
-     * target VM was unable to pop the frames.
+     * @throws OpaqueFrameException if target VM is unable to pop this frame
+     * (e.g. a virtual thread is not suspended at an event).
```

Comment:
Nit: I'm not sure the comment at line 406 is fully correct. The `IncompatibleThreadStateException` will be thrown if a virtual thread is not suspended. So, the `OpaqueFrameException` will be thrown only when a virtual thread is suspended but not at an event. The same concern should apply for the line  488.

Reply:
> Maybe it should just say "suspended, but not at an event"?

Yes, this would make it clear.
And yes, the JVMTI vs JDI spec difference is reasonable.